api/src/services/bctw-service.ts

```diff
@@ -161,10 +165,39 @@ export class BctwService {
     return await this.axiosInstance.post(DEPLOY_DEVICE_ENDPOINT, device);
   }
 
-  async getDeviceDetails(deviceId: number): Promise<Record<string, unknown>[]> {
+  /**
+   * Update device hardware details in BCTW.
+   *
+   * @param {IDevice} device
+   * @returns {*} {IDevice}
+   * @memberof BctwService
+   */
+  async updateDevice(device: IDevice): Promise<IDevice> {
+    const { data } = await this.axiosInstance.post(UPSERT_DEVICE_ENDPOINT, device);
+    if (data.errors.length) {
+      throw Error(JSON.stringify(data.errors));
+    }
+    return data;
+  }
+
+  /**
+   * Get device hardware details by device id.
+   *
+   * @param deviceId
+   * @returns {*} {Promise<IDevice[]>}
+   * @memberof BctwService
+   */
+  async getDeviceDetails(deviceId: number): Promise<IDevice[]> {
     return await this._makeGetRequest(`${GET_DEVICE_DETAILS}${deviceId}`);
```

Comment:
Nit: `await` here is redundant (and technically slower). Just return the promise